File: Searching/PythonCodes/Leetcode744.py
class Solution(object):
    def nextGreatestLetter(self, letters, target):

        start, end = 0, len(letters) - 1

        while start < end:
            mid = start + (end - start) // 2
            if target >= letters[mid]:
                start = mid + 1
            else:
                end = mid
        
        return letters[start] if letters[start] > target else letters[0]

File: Searching/JavaCodes/Leetcode744.java
class Solution {
    public static char nextGreatestLetter(char[] letters, char target) {
        int start=0,end=letters.length-1;

        while(start<end){
            int mid=start+(end-start)/2;
            if(target>=letters[mid]){
                start=mid+1;
            }
            else{
                end=mid;
            }
        }
        return letters[start]>target?letters[start]:letters[0];
    }
}